### PR TITLE
Add support for MarkDown headings formatted as "COMMAND(1) ..."

### PR DESCRIPTION
**Pull request by thaJeztah**
relates to https://github.com/spf13/cobra/issues/1049 and https://github.com/docker/docker-ce-packaging/pull/502

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @cpuguy83 @tianon PTAL

**Comment by thaJeztah** [comment]
hm, actually, more may be needed for cobra, as I think it's using `% COMMAND(1)` (so  not a heading)

File: md2man/roff_test.go [hunk 1]
```diff
@@ -305,8 +333,12 @@ func execRecoverableTestSuite(t *testing.T, tests []string, params TestParams, s
 
 func runMarkdown(input string, params TestParams) string {
 	renderer := NewRoffRenderer()
+	extensions := params.extensions
+	if extensions == 0 {
+		extensions = renderer.extensions
+	}
```
Comment (thaJeztah) [inline md2man/roff_test.go:339]: So, after digging why it didn't work with the `% xxx` pandoc headers, this seemed to be the problem; the tests were running without the `blackfriday.Titleblock` extension enabled, so blackfriday was not converting them to H1 blocks.

**Comment by thaJeztah** [comment]
@cpuguy83 @tianon this should be ready to go now. I noticed that v1 would convert this input;

```markdown
% DOCKER(1) Docker User Manuals
% Docker Community
% JUNE 2014
```

To:

```
.TH "DOCKER" "1" " Docker User Manuals" "Docker Community" "JUNE 2014"
.nh
.ad l
```

And v2 (with this change applied) converts it to:

```
.nh
.TH DOCKER 1 Docker User Manuals
Docker Community
JUNE 2014
```

As a result, the words were split, and didn't show up correctly when viewing the man page;


With V1:

```
man ./v1.1

DOCKER(1)                                                                              JUNE 2014                                                                              DOCKER(1)

Docker Community                                                                   Docker User Manuals                                                                        DOCKER(1)
```

With V2:

```
man ./v2.1

DOCKER(1)                                                                               Manuals                                                                               DOCKER(1)

Docker Community JUNE 2014

User                                                                                     Docker                                                                               DOCKER(1)
```

I pushed another commit to remove newlines, and to quote each line. It's  a bit ugly, so better suggestions welcome

**Comment by cpuguy83** [comment]
Is this still needed?

**Comment by cpuguy83** [comment]
@thaJeztah ?

**Comment by thaJeztah** [comment]
I guess with the fix in Cobra, it's no longer needed; let me close this one (I'll keep the branch for now, in case we find we still need it)

**Pull request closed without merging**